### PR TITLE
upgrade mongodb driver

### PR DESCRIPTION
**Pull request by naderio**
- upgrade mongodb driver to `^2.2.22` (to support MongoDB 3.6) and update connection options
- cleanup/fix `find()` and `destroy()`
- drops support for Node.js 0.10 since [it breaks tests](https://ci.appveyor.com/project/mikermcneil/sails-mongo/build/1.0.171/job/3k7550l81nke0u2t)
- add Node.js 8, 9 and 10 to CI

### CODE REVIEW

**Comment by mikermcneil** [comment]
@naderio would you mind rebasing this to master?

**Comment by AtiX** [comment]
Any chance @naderio you could get this ready to be merged. I'd highly appreciate it!

**Comment by naderio** [comment]
@mikermcneil this PR is patch intended for version 0.12.x (branch 0.12.x) and NOT for 1.x (branch master) .. therefore rebasing against master is not needed
@AtiX PR is ready to be merged

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # Sails-Mongo Changelog
 
+### 0.12.4
+
+* [INTERNAL] Bump `mongodb` version for
```
Comment (patrick-webs) [inline CHANGELOG.md]: This sentence is not finished

**Comment by patrick-webs** [comment]
@mikermcneil Is this likely/unlikely to be merged at some point and released as 0.12.4?

I'm investigating updating a sails 0.12 app to work with newer MongoDB versions and hoping to avoid having to upgrade to sails 1.x at the same time, another 0.12.x release would be great for us.

MongoDB 3.0 and 3.2 support has been EOLed as of this month and [Mongo's driver documentation ](https://docs.mongodb.com/ecosystem/drivers/driver-compatibility-reference/#node-js-driver-compatibility) states that a driver >=2.2.12 is recommended to use MongoDB 3.4.

**Comment by mikermcneil** [comment]
cc @luislobo

**Comment by luislobo** [comment]
@patrick-webs @mikermcneil I actually have a fork with a branch that has updated mongodb drivers that work with Atlas (we are using it ourselves).
I can create a PR, and submit it.

And in the near future, I plan to evaluate migrating sails-mongo to the latest native mongodb driver. But at least the one I have works.
Check if our branch works for you https://github.com/PetroCloud/sails-mongo/tree/0-13

**Comment by luislobo** [comment]
I wll compare this PR and our branch to see if we can join efforts.

**Comment by naderio** [comment]
FYI I have this version running for months in production with MongoDB 3.6 and no issues.
There was a number of warning that got removed after updating connection options.

**Comment by luislobo** [comment]
@naderio yeah, my branch had the same issues. I'l double check mine and yours, and see if it is fully compatible with MongoDB Atlas

**Comment by patrick-webs** [comment]
FYI @naderio , I tried using your branch, it worked fine to connect to a local MongoDB 3.4 instance. When I tried connecting to a MongoDB 3.4 on Atlas, I began to get issues causing sails to not start:

    MongoError: connection 8 to obfuscated-shard-00-02-dmqao.mongodb.net:27017 closed

I found that I had to add a `ssl: true` option to the `config/connections.js` file which had not been required before; my connection string already had `ssl=true` in it. It seems like something about the upgrade made that query string parameter in the connection string get ignored. I don't know if it's code in the branch, or if it was a change in the mongodb driver, but figured I'd document it here in case anyone else runs into the same issues.

**Comment by luislobo** [comment]
@patrick-webs Yeah, we had to do the same here too. I'll try to get some time extra to review this PR during the weekend.

**Comment by luislobo** [comment]
@naderio I've added several comments in the PR's code. To add to those, what @patrick-webs mentions should be added to the README.md too, as an "atlas" configuration section.

**Comment by fabdrol** [comment]
@naderio @luislobo what's the ETA for this? We're running into the same problem.

**Comment by naderio** [comment]
@luislobo @patrick-webs ssl issue should be fixed, tested with mongodb atlas

**Comment by constantx** [comment]
any chance for this to be merged soon?

**Comment by simonmilz** [comment]
I just received an email from mLab that they upgrade all mongo instances to 3.6 by May 7th. Any chance to get this PR merged soon?

**Comment by patrick-webs** [comment]
FWIW, I privately forked https://github.com/websdev/sails-mongo/commit/97e6a4f5033f765fb08fa06de134f7e5f9d71ff5 and published it to a private npm module for my use, we've been using it in production with MongoDB Atlas without issue for the last 6 months.

**Comment by simonmilz** [comment]
@patrick-webs can you please share you implementation? The mentioned repo is private.

**Comment by simonmilz** [comment]
I applied the fixes here: https://github.com/TypischTouristik/sails-mongo

**Comment by patrick-webs** [comment]
@simonmilz Sorry, I referenced the wrong repo in my comment. I updated my comment to reference the commit from this pull request that I meant to reference. The fork I am using has no other changes besides private configuration files for building and uploading to my private npm repository.

**Comment by manuelzi** [comment]
> I applied the fixes here: https://github.com/TypischTouristik/sails-mongo

Thanks for sharing this. Has anyone tried to upgrade the driver on the legacy branch to >=3.1 for MongoDB 4.0 support?

**Comment by simonmilz** [comment]
@manuelzi we are working on this right now.

**Comment by manuelzi** [comment]
@simonmilz Ok thanks, let me know if you need help testing.

**Comment by DominusKelvin** [comment]
Hey @naderio, @AtiX, @simonmilz, @manuelzi, @constantx, and, @patrick-webs, I'll be closing this PR as this other [PR](https://github.com/balderdashy/sails-mongo/pull/499) has implemented the upgrade and it has been merged. Thank you for working on this  @naderio 💙

Also you can check here for the upgrade information. 👇🏾

https://blog.sailscasts.com/sails-mongo-v2-1-0

**Pull request closed without merging**